File: app/src/main/java/com/example/contact/view/MainActivity.java
package com.example.contact.view;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.database.Cursor;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ImageButton;
import android.widget.ListView;
import android.widget.SearchView;
import android.widget.TextView;

import com.example.contact.R;
import com.example.contact.controller.CustomAdapter;
import com.example.contact.controller.SQLiteController;
import com.example.contact.model.ContactItem;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    //attributes
private ListView listView;
private ArrayList<ContactItem> arrayList;
private ArrayList arrayListDoublon;
private ArrayAdapter arrayAdapterDoublon;
private CustomAdapter arrayAdapter;
private ImageButton ajouterContact, infoImageButton;
private ContactItem contactItem;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listView = (ListView) findViewById(R.id.listView);
        ajouterContact = (ImageButton) findViewById(R.id.addButton);
        infoImageButton = (ImageButton) findViewById(R.id.imageButtonInformation) ;



        ajouterContact.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), InsertContact.class);
                startActivity(intent);
                finish();
            }
        });

        arrayList = new ArrayList<>();
        arrayListDoublon = new ArrayList();

        SQLiteController sqLiteController = new SQLiteController(getApplicationContext());
//        sqLiteController.InsertContact(new ContactItem(1,"Romulus", "Ronick","38471151", "34, rue casseus", "roromulus@yahoo.com"));
        Cursor cursor = sqLiteController.selectContact();
        while(cursor.moveToNext()){
            Integer id = cursor.getInt(cursor.getColumnIndex("id"));
            String nom = cursor.getString(cursor.getColumnIndex("nom"));
            String prenom = cursor.getString(cursor.getColumnIndex("prenom"));
            String phone = cursor.getString(cursor.getColumnIndex("phone"));
            String adresse = cursor.getString(cursor.getColumnIndex("adresse"));
            String email = cursor.getString(cursor.getColumnIndex("email"));

            contactItem = new ContactItem(id, nom, prenom,phone,adresse,email);
            arrayList.add(contactItem);
            arrayListDoublon.add(nom + " "+prenom);
//            arrayListDoublon.set(id, prenom+" "+nom);


        }
        arrayAdapterDoublon = new ArrayAdapter(getApplicationContext(),android.R.layout.simple_list_item_1,arrayListDoublon);
        arrayAdapter = new CustomAdapter(getApplicationContext(), R.layout.customlist,arrayList);
        listView.setAdapter(arrayAdapterDoublon);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

                //on va recuperer id du contact selectionne
                Intent intent = new Intent(getApplicationContext(), AfficherContact.class);
                String nomFiltre= arrayAdapterDoublon.getItem(position).toString();
                int identifiant = findIdByName(arrayList, nomFiltre);
//                String tempon = String.valueOf(id);
//                Integer resultat = Integer.parseInt(tempon);
                intent.putExtra("id", identifiant );
                startActivity(intent);
//                finish();
            }
        });

        infoImageButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AlertDialog.Builder builder1 = new AlertDialog.Builder(MainActivity.this);
                builder1.setTitle("Infos Contact");
                String message ="Il y a "+ arrayListDoublon.size()+" contacts enregistres";
                builder1.setMessage(message);
                builder1.setCancelable(true);
                builder1.setNeutralButton(android.R.string.ok,
                        new DialogInterface.OnClickListener() {
                            public void onClick(DialogInterface dialog, int id) {
                                dialog.cancel();
                            }
                        });

                AlertDialog alert11 = builder1.create();
                alert11.show();
            }
        });

    }

    //Implementation de search bar
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.menu_item, menu);
        MenuItem menuItem = menu.findItem(R.id.app_bar_search);

        //filtering
        SearchView searchView = (SearchView) menuItem.getActionView();

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                arrayAdapterDoublon.getFilter().filter(query);
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                arrayAdapterDoublon.getFilter().filter(newText);
                return false;
            }
        });

        return true;
    }

    //methode pour comparer le nom resultant d une recherche avec les noms contenus ds les objets afin d en trouver l'id
    public int findIdByName(ArrayList<ContactItem> table, String name){
        int i, idObject=-1;
        for(i=0; i<table.size(); i++ ){

            String nameObject = table.get(i).getNom()+" "+table.get(i).getPrenom();
            if(name.equals(nameObject)){
                idObject = table.get(i).getId();
            }
        }
        return idObject;
    }
}

File: app/src/main/java/com/example/contact/view/MainActivity.kt
package com.example.contact.view

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.ImageButton
import android.widget.ListView
import android.widget.SearchView
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.Volley
import com.example.contact.R
import com.example.contact.controller.CustomAdapter
import com.example.contact.controller.SQLiteController
import com.example.contact.model.ContactItem
import java.util.*
import kotlin.collections.ArrayList

class MainActivity : AppCompatActivity() {
    //attributes
    lateinit var listView: ListView
     val arrayList = ArrayList<ContactItem>()
    lateinit var resultats: ArrayList<ContactItem>
    lateinit var arrayListDoublon: ArrayList<String>
//    private val arrayAdapterDoublon: ArrayAdapter<*>? = null
    companion object{
    lateinit var arrayAdapter: CustomAdapter
}

    lateinit var ajouterContact: ImageButton
    lateinit var infoImageButton: ImageButton

    lateinit var requestQ: RequestQueue

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        listView = findViewById<View>(R.id.listView) as ListView
        ajouterContact = findViewById<View>(R.id.addButton) as ImageButton
        infoImageButton = findViewById<View>(R.id.imageButtonInformation) as ImageButton
        ajouterContact.setOnClickListener {
            val intent = Intent(applicationContext, InsertContact::class.java)
            startActivity(intent)
            finish()
        }

        arrayListDoublon = ArrayList()

        requestQ = Volley.newRequestQueue(applicationContext)

        val url = "http://192.168.0.103/contact/manager.php?action=select"
        val jsonObjectRequest = JsonObjectRequest(Request.Method.GET,url,null,
                { response ->
                    var jsonArray = response.getJSONArray("contact_tbl")
                    for (i in 0 until jsonArray.length()){
                        var jsonObject = jsonArray.getJSONObject(i)
                        var id = jsonObject.getInt("id")
                        var prenom = jsonObject.getString("prenom")
                        var nom = jsonObject.getString("nom")
                        var adresse = jsonObject.getString("adresse")
                        var email= jsonObject.getString("email")
                        var phone = jsonObject.getString("phone")
                        // Toast.makeText(context, prenom+" "+ nom, Toast.LENGTH_LONG).show()

                       var contactItem = ContactItem(id, nom, prenom, phone, adresse, email)
                        arrayList.add(contactItem)
                        arrayAdapter.notifyDataSetChanged()
                        println(arrayList[i].nom)

                        arrayListDoublon.add("$nom $prenom")
                    }

                },
                {
                    error ->
                    println("There is an error: "+ error)

                })
        requestQ.add(jsonObjectRequest)

        arrayAdapter = CustomAdapter(applicationContext, R.layout.customlist, arrayList)
        listView.adapter = arrayAdapter


        listView.onItemClickListener = OnItemClickListener { parent, view, position, id ->
            //on va recuperer id du contact selectionne
            val intent = Intent(applicationContext, AfficherContact::class.java)
            //                String nomFiltre= arrayAdapterDoublon.getItem(position).toString();
//                int identifiant = findIdByName(arrayList, nomFiltre);

//                String tempon = String.valueOf(id);
//                Integer resultat = Integer.parseInt(tempon);
            val identifiant = arrayAdapter.getItem(position)!!.id
            val nom = arrayAdapter.getItem(position)!!.nom
            val prenom = arrayAdapter.getItem(position)!!.prenom
            val adresse = arrayAdapter.getItem(position)!!.adresse
            val phone = arrayAdapter.getItem(position)!!.phone
            val email = arrayAdapter.getItem(position)!!.email
            intent.putExtra("id", identifiant)
            intent.putExtra("nom", nom)
            intent.putExtra("prenom", prenom)
            intent.putExtra("adresse", adresse)
            intent.putExtra("phone", phone)
            intent.putExtra("email", email)
            startActivity(intent)
            //                finish();
        }
        infoImageButton.setOnClickListener {
            val builder1 = AlertDialog.Builder(this@MainActivity)
            builder1.setTitle("Infos Contact")
            val message = "Il y a " + arrayListDoublon.size + " contacts enregistres"
            builder1.setMessage(message)
            builder1.setCancelable(true)
            builder1.setNeutralButton(android.R.string.ok
            ) { dialog, id -> dialog.cancel() }
            val alert11 = builder1.create()
            alert11.show()
        }
    }

    //Implementation de search bar
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val menuInflater = menuInflater
        menuInflater.inflate(R.menu.menu_item, menu)
        val menuItem = menu.findItem(R.id.app_bar_search)

        //filtering
        val searchView = menuItem.actionView as SearchView
        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                return false
            }

            override fun onQueryTextChange(newText: String): Boolean {
                var nom: String
                var prenom: String
                var completeName: String
                val filtre: String
                filtre = newText.replace("\\s+".toRegex(), "").toLowerCase()
                resultats = ArrayList()
                for (x in arrayList) {
                    nom = x.nom.toLowerCase()
                    prenom = x.prenom.toLowerCase()
                    completeName = prenom + nom
                    if (nom.contains(newText.toLowerCase()) or prenom.contains(newText.toLowerCase()) or completeName.contains(filtre)) {
                        resultats.add(x)
                    }
                }
                (listView.adapter as CustomAdapter).update(resultats)
                return false
            }
        })
        return true
    }

    //methode pour comparer le nom resultant d une recherche avec les noms contenus ds les objets afin d en trouver l'id
    fun findIdByName(table: ArrayList<ContactItem>, name: String): Int {
        var i: Int
        var idObject = -1
        i = 0
        while (i < table.size) {
            val nameObject = table[i].nom + " " + table[i].prenom
            if (name == nameObject) {
                idObject = table[i].id
            }
            i++
        }
        return idObject
    }
}